src/common/src/array/arrow/arrow_impl.rs

```diff
@@ -692,6 +709,33 @@ pub trait FromArrow {
         Ok(ArrayImpl::Time(array.into()))
     }
 
+    fn from_timestampsecond_array(
+        &self,
+        array: &arrow_array::TimestampSecondArray,
+    ) -> Result<ArrayImpl, ArrayError> {
+        Ok(ArrayImpl::Timestamp(array.into()))
+    }
+    fn from_timestampsecond_some_array(
+        &self,
+        array: &arrow_array::TimestampSecondArray,
+    ) -> Result<ArrayImpl, ArrayError> {
+        Ok(ArrayImpl::Timestamp(array.into()))
+    }
+
+    fn from_timestampms_array(
+        &self,
+        array: &arrow_array::TimestampMillisecondArray,
+    ) -> Result<ArrayImpl, ArrayError> {
+        Ok(ArrayImpl::Timestamp(array.into()))
+    }
+
+    fn from_timestampms_some_array(
+        &self,
+        array: &arrow_array::TimestampMillisecondArray,
+    ) -> Result<ArrayImpl, ArrayError> {
+        Ok(ArrayImpl::Timestamptz(array.into()))
+    }
```

Comment:
I am curious, we don't need to do any conversion between these types? All of them can be implemented in the same way?

Reply:
No, just added such logic and verify it works. I constructed a parquet file containing timestamps of four time units, then imported it into RisingWave and it was parsed successfully:
<img width="1207" alt="image" src="https://github.com/user-attachments/assets/e1dbe4d7-3757-4194-bf7b-cf1c0756494a">

Reply:
I also don't get why we need to add different methods to convert timestamp in different units. IIUC, the current conversion [logic ](https://github.com/risingwavelabs/risingwave/blob/fdeeb2e6c358fe8f04d1fa29ef5384e75ae2e3f8/src/common/src/array/arrow/arrow_impl.rs#L944)won't lose any precision. What happened if we just add different units of the timestamp type to `converts` macro and rely on the original [implementation ](https://github.com/risingwavelabs/risingwave/blob/fdeeb2e6c358fe8f04d1fa29ef5384e75ae2e3f8/src/common/src/array/arrow/arrow_impl.rs#L942) to do the conversion?
```
converts!(TimestampArray, arrow_array::TimestampMillisecondArray, @map);
converts!(TimestampArray, arrow_array::TimestampSecondArray, @map);
converts!(TimestampArray, arrow_array::TimestampNanosecondArray, @map);
```

Reply:
In our previous code, we only handled timestamps in microseconds, so this conversion is only correct for `timestamp(MicroSecond, _)`. As you mentioned in the [logic](https://github.com/risingwavelabs/risingwave/blob/fdeeb2e6c358fe8f04d1fa29ef5384e75ae2e3f8/src/common/src/array/arrow/arrow_impl.rs#L944), the seconds part is `% 1_000_000`, and the nanoseconds part is also `% 1_000_000`. If all four time units are converted using this logic, it will yield incorrect results, as follows:
<img width="1126" alt="image" src="https://github.com/user-attachments/assets/82b7b714-6c88-4b14-9586-b5f42a602615">

Reply:
The correct approach should be as shown in the PR, where different handling is done based on the time unit. For example, for milliseconds, it should be `%1000` instead of `%100000`:
```
 TimeUnit::Millisecond => Timestamp(
                DateTime::from_timestamp((value / 1_000) as _, (value % 1_000 * 1000000) as _)
                    .unwrap()
                    .naive_utc(),
            ),
```

So I think specifying the time unit is essential.